pkg/upload/android-proguard.go

```diff
@@ -1,15 +1,127 @@
 package upload
 
 import (
+	"path/filepath"
+	"strconv"
+
+	"github.com/bugsnag/bugsnag-cli/pkg/log"
+	"github.com/bugsnag/bugsnag-cli/pkg/server"
 	"github.com/bugsnag/bugsnag-cli/pkg/utils"
 )
 
 type AndroidProguardMapping struct {
 	ApplicationId   string            `help:"Module application identifier"`
-	AppManifestPath utils.UploadPaths `help:"(required) Path to directory or file to upload"`
+	AppManifestPath string            `help:"(required) Path to directory or file to upload" type:"path"`
 	BuildUuid       string            `help:"Module Build UUID"`
 	Configuration   string            `help:"Build type, like 'debug' or 'release'"`
 	Path            utils.UploadPaths `arg:"" name:"path" help:"(required) Path to directory or file to upload" type:"path"`
 	VersionCode     string            `help:"Module version code"`
 	VersionName     string            `help:"Module version name"`
+	DryRun          bool              `help:"Validate but do not upload"`
+}
+
+func ProcessAndroidProguard(paths []string, applicationId string, appManifestPath string, buildUuid string, configuration string, versionCode string, versionName string, endpoint string, timeout int, retries int, overwrite bool, apiKey string, failOnUploadError bool, dryRun bool) error {
+
+	uploadFileOptions := make(map[string]map[string]string)
+
+	for _, path := range paths {
+		// Path is a directory
+		if utils.IsDir(path) {
+			mergedManifestsPath := filepath.Join(path, "build", "intermediates", "merged_manifests")
+			if utils.IsDir(mergedManifestsPath) {
+				variants, err := utils.BuildVariantsList(mergedManifestsPath)
+				if err != nil {
+					log.Error(err.Error(), 1)
+				}
+
+				for _, variant := range variants {
+					uploadFileOptions[variant] = map[string]string{}
+					uploadFileOptions[variant]["androidManifestPath"] = filepath.Join(mergedManifestsPath, variant, "AndroidManifest.xml")
+					uploadFileOptions[variant]["mappingPath"] = filepath.Join(mergedManifestsPath, "..", "..", "outputs", "mapping", variant, "mapping.txt")
+
+				}
```

Comment:
nit: random blank line?
```suggestion
					uploadFileOptions[variant]["mappingPath"] = filepath.Join(mergedManifestsPath, "..", "..", "outputs", "mapping", variant, "mapping.txt")
				}
```